diff --git c/packages/tailwindcss/src/at-import.ts i/packages/tailwindcss/src/at-import.ts
index 5f13cce9..0e9b8b25 100644
--- c/packages/tailwindcss/src/at-import.ts
+++ i/packages/tailwindcss/src/at-import.ts
@@ -3,7 +3,7 @@ import { atRule, context, walk, WalkAction, type AstNode } from './ast'
 import * as CSS from './css-parser'
 import * as ValueParser from './value-parser'
 
-type LoadStylesheet = (
+export type LoadStylesheet = (
   id: string,
   basedir: string,
 ) => Promise<{
diff --git c/packages/tailwindcss/src/patches.ts i/packages/tailwindcss/src/patches.ts
new file mode 100644
index 00000000..24e324e4
--- /dev/null
+++ i/packages/tailwindcss/src/patches.ts
@@ -0,0 +1,91 @@
+import type { Candidate } from "./candidate"
+import type { DesignSystem } from "./design-system"
+import type { Theme } from "./theme"
+
+const REGEX = /^(([a-z]+)|(\d+))(-(\d{1,3})-(\d{1,3}))?(-(\d{1,2}|100))?$/i
+
+const parseNumber = (value: string | undefined) =>
+  value === undefined ? undefined : parseInt(value, 10)
+
+export const isColor = (
+  variable: string,
+  theme: Theme
+): boolean => {
+  const array = REGEX.exec(variable)
+
+  if (array === null) {
+    return false
+  }
+
+  const colorString = array[2] as string | undefined
+  const colorNumber = colorString === undefined ? parseNumber(array[3])! : undefined
+  const color = (colorString ?? colorNumber)
+
+  if (color === undefined) {
+    return false
+  }
+
+  if (typeof color === 'string' && theme.colorReferences.has(color)) {
+    return true
+  }
+
+  const lightness = array[5]
+  const chroma = lightness === undefined ? undefined : array[6] as string | undefined
+
+  if (lightness === undefined || chroma === undefined) {
+    return false
+  }
+
+  return true
+}
+
+export const resolveColorReference = (value: Exclude<Extract<Candidate, { kind: 'functional' }>['value'], null>, theme: Theme) => {
+  if (value.kind === 'named' && isColor(value.value, theme)) {
+    return `var(---color-${value.value})`
+  }
+
+  return null
+}
+
+export type MapLazyResolver<K, V> = (key: K) => V | undefined
+export type ThemeEntry = ReturnType<DesignSystem['theme']['values']['entries']> extends MapIterator<infer T> ? T : never
+export type ThemeEntryKey = ThemeEntry[0]
+export type ThemeEntryValue = ThemeEntry[1]
+export type ThemeValueResolver = MapLazyResolver<ThemeEntryKey, ThemeEntryValue>
+
+export class MapLazy<K, V> extends Map<K, V> {
+  private readonly resolvers: MapLazyResolver<K, V>[] = [];
+
+  constructor()
+  constructor(
+    entries?: readonly (readonly [K, V])[] | null,
+    resolvers?: MapLazyResolver<K, V>[],
+  )
+  constructor(
+    entries?: readonly (readonly [K, V])[] | null,
+    resolvers: MapLazyResolver<K, V>[] = [],
+  ) {
+    super(entries ?? undefined)
+    this.resolvers = resolvers
+  }
+
+  private tryResolve(key: K): V | undefined {
+    for (const resolve of this.resolvers) {
+      const value = resolve(key)
+      if (value !== undefined) {
+        super.set(key, value as V)
+        return value as V
+      }
+    }
+
+    return undefined
+  }
+
+  override has(key: K): boolean {
+    return super.has(key) || this.tryResolve(key) !== undefined
+  }
+
+  override get(key: K): V | undefined {
+    return super.has(key) ? super.get(key) : this.tryResolve(key)
+  }
+}
diff --git c/packages/tailwindcss/src/theme.ts i/packages/tailwindcss/src/theme.ts
index dc9c750a..08a64966 100644
--- c/packages/tailwindcss/src/theme.ts
+++ i/packages/tailwindcss/src/theme.ts
@@ -1,4 +1,5 @@
 import { type AtRule, type Declaration } from './ast'
+import { MapLazy, type ThemeValueResolver } from './patches'
 import { escape, unescape } from './utils/escape'
 
 export const enum ThemeOptions {
@@ -38,9 +39,11 @@ function isIgnoredThemeKey(themeKey: ThemeKey, namespace: ThemeKey) {
 
 export class Theme {
   public prefix: string | null = null
+  public colorReferences = new Set<string>()
+  public resolvers: ThemeValueResolver[] = []
 
   constructor(
-    private values = new Map<
+    public values = new MapLazy<
       string,
       {
         value: string
@@ -48,7 +51,7 @@ export class Theme {
         src: Declaration['src']
       }
     >(),
-    private keyframes = new Set<AtRule>([]),
+    public keyframes = new Set<AtRule>([]),
   ) {}
 
   get size() {
diff --git c/packages/tailwindcss/src/utilities.ts i/packages/tailwindcss/src/utilities.ts
index 0fdb5c9f..2eeaf81f 100644
--- c/packages/tailwindcss/src/utilities.ts
+++ i/packages/tailwindcss/src/utilities.ts
@@ -26,6 +26,8 @@ import { replaceShadowColors } from './utils/replace-shadow-colors'
 import { segment } from './utils/segment'
 import * as ValueParser from './value-parser'
 
+import { resolveColorReference } from './patches'
+
 const IS_VALID_STATIC_UTILITY_NAME = /^-?[a-z][a-zA-Z0-9/%._-]*$/
 const IS_VALID_FUNCTIONAL_UTILITY_NAME = /^-?[a-z][a-zA-Z0-9/%._-]*-\*$/
 
@@ -70,13 +72,13 @@ type CompileFn<T extends Candidate['kind']> = (
   value: Extract<Candidate, { kind: T }>,
 ) => AstNode[] | undefined | null
 
-interface SuggestionGroup {
+export interface SuggestionGroup {
   supportsNegative?: boolean
   values: (string | null)[]
   modifiers: string[]
 }
 
-type SuggestionDefinition =
+export type SuggestionDefinition =
   | string
   | {
       supportsNegative?: boolean
@@ -232,7 +234,7 @@ export function asColor(
  * The values `inherit`, `transparent` and `current` are special-cased as they
  * are universal and don't need to be resolved from the theme.
  */
-function resolveThemeColor<T extends ThemeKey>(
+export function resolveThemeColor<T extends ThemeKey>(
   candidate: Extract<Candidate, { kind: 'functional' }>,
   theme: Theme,
   themeKeys: T[],
@@ -264,6 +266,11 @@ function resolveThemeColor<T extends ThemeKey>(
     }
     default: {
       value = theme.resolve(candidate.value!.value, themeKeys)
+
+      if (value === null) {
+        value = resolveColorReference(candidate.value!, theme)
+      }
+
       break
     }
   }
@@ -280,6 +287,16 @@ function resolveThemeColor<T extends ThemeKey>(
  */
 const LEGACY_NUMERIC_KEY = /(\d+)_(\d+)/g
 
+export type UtilityDescription = {
+  supportsNegative?: boolean
+  supportsFractions?: boolean
+  themeKeys?: ThemeKey[]
+  defaultValue?: string | null
+  handleBareValue?: (value: NamedUtilityValue) => string | null
+  handleNegativeBareValue?: (value: NamedUtilityValue) => string | null
+  handle: (value: string, dataType: string | null) => AstNode[] | undefined
+}
+
 export function createUtilities(theme: Theme) {
   let utilities = new Utilities()
 
@@ -366,16 +383,6 @@ export function createUtilities(theme: Theme) {
     })
   }
 
-  type UtilityDescription = {
-    supportsNegative?: boolean
-    supportsFractions?: boolean
-    themeKeys?: ThemeKey[]
-    defaultValue?: string | null
-    handleBareValue?: (value: NamedUtilityValue) => string | null
-    handleNegativeBareValue?: (value: NamedUtilityValue) => string | null
-    handle: (value: string, dataType: string | null) => AstNode[] | undefined
-  }
-
   /**
    * Register a functional utility class like `max-w-*` that maps to tokens in the
    * user's theme.
@@ -5725,6 +5732,14 @@ export function createUtilities(theme: Theme) {
     },
   ])
 
+  Object.assign(utilities, {
+    references: {
+      functionalUtility,
+      staticUtility,
+      suggest
+    }
+  })
+
   return utilities
 }
 
diff --git c/packages/tailwindcss/src/utils/infer-data-type.ts i/packages/tailwindcss/src/utils/infer-data-type.ts
index 85b99c6d..36dc53c8 100644
--- c/packages/tailwindcss/src/utils/infer-data-type.ts
+++ i/packages/tailwindcss/src/utils/infer-data-type.ts
@@ -365,7 +365,7 @@ export function isValidOpacityValue(value: any) {
  * Ensures a number (or numeric string) is a multiple of another number, and
  * that it has no unnecessary leading or trailing zeros.
  */
-function isMultipleOf(value: string | number, divisor: number) {
+export function isMultipleOf(value: string | number, divisor: number) {
   let num = Number(value)
   return num >= 0 && num % divisor === 0 && String(num) === String(value)
 }
diff --git c/packages/tailwindcss/src/variants.ts i/packages/tailwindcss/src/variants.ts
index 701f08c0..30c56224 100644
--- c/packages/tailwindcss/src/variants.ts
+++ i/packages/tailwindcss/src/variants.ts
@@ -1132,7 +1132,7 @@ export function createVariants(theme: Theme): Variants {
   staticVariant('ltr', ['&:where(:dir(ltr), [dir="ltr"], [dir="ltr"] *)'])
   staticVariant('rtl', ['&:where(:dir(rtl), [dir="rtl"], [dir="rtl"] *)'])
 
-  staticVariant('dark', ['@media (prefers-color-scheme: dark)'])
+  // staticVariant('dark', ['@media (prefers-color-scheme: dark)'])
 
   staticVariant('starting', ['@starting-style'])
 
